src/main/scala/freeeventsourcing/syntax/ProcessSyntax.scala

```diff
@@ -0,0 +1,218 @@
+  def waitUntil(when: Instant) =
+    lift[Unit](WaitUntil[BC](when))
+
+  /** Wait for multiple events and run the path of the first event. */
+  def firstOf[Paths <: HList](b: FirstOfBuilder[HNil] ⇒ FirstOfBuilder[Paths])(implicit switch: Switch[Paths]): ProcessMonad[switch.Result] = {
+    val paths = b(new FirstOfBuilder(HNil)).collect
+    val alternatives = switch.alternatives(paths)
+    lift[alternatives.Events](FirstOf[BC, alternatives.type](alternatives))
+      .flatMap(e ⇒ switch.effectFor(paths)(e))
+  }
+
+  /** Wait for multiple events and run the path of the first event. */
+  def firstOfUnified[Paths <: HList, R <: Coproduct](b: FirstOfBuilder[HNil] ⇒ FirstOfBuilder[Paths])(implicit s: Switch.Aux[Paths, R], u: Unifier[R]) =
+    firstOf(b).map(r ⇒ u(r))
+
+  /** Terminate this process instance. */
+  def terminate = lift[Unit](End[BC]())
+
+  /** Does nothing. */
+  def noop = Monad[ProcessMonad].pure(())
+
+  /////////////////////////////////////////////////////
+
+  private def lift[A](action: ProcessAction[BC, A]) = Free.liftF[ProcessAction[BC, ?], A](action)
+
+  /** Internas used by the above functions. Need to be public but are not meant to be used directly. */
+  object Builders {
+    /** Helper class for on. */
+    final class OnAggregateBuilder[A <: Aggregate] private[ProcessSyntax] (aggregateType: A, aggregate: A#Id) {
+      /** Execute a command on the aggregate. */
+      def execute[R <: Coproduct](command: A#Command)(
+        catches: CommandErrorHandler.EmptyBuilder[command.Error] ⇒ CommandErrorHandler.Builder[command.Error, R]
+      )(implicit ev: ValidAggregate[BC, A], ev2: AllErrorsHandled[R]) = {
+        val builder = CommandErrorHandler.builder[command.Error]
+        val errorHandler = catches(builder).errorHandler
+        lift[Unit](Execute[BC, A, command.type](aggregateType, aggregate, command, errorHandler))
+      }
+    }
+
+    /** Helper class for from. */
+    final class FromAggregateBuilder[A <: Aggregate] private[ProcessSyntax] (aggregateType: A, aggregate: A#Id) {
+      /** Await an event from the aggregate. */
+      def await[E <: A#Event: AggregateEventType](implicit ev: ValidAggregate[BC, A], idser: StringSerializable[A#Id]) = {
+        val selector = AggregateEventSelector(aggregateType)(aggregate)[E]
+        ProcessSyntax.this.await(selector)
+      }
+    }
+
+    /** Helper class for execute. */
+    type CommandErrorHandler[Error <: Coproduct] = Error ⇒ ProcessMonad[Unit]
+    object CommandErrorHandler {
+      private[ProcessSyntax] def builder[E <: Coproduct]() = new Builder[E, E](e ⇒
+        throw new AssertionError("Error in CommandErrorHandler, type was constructed that does not " +
+          "handle all cases. Should have been prevented by the compiler"))
+
+      type EmptyBuilder[E <: Coproduct] = Builder[E, E]
+
+      final class Builder[All <: Coproduct, Unhandled <: Coproduct] private[CommandErrorHandler] (handled: CommandErrorHandler[All]) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class name does not match the regular expression '"[A-Z][A-Za-z]*"'](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2571398663/issues/source?bid=3298199&fileBranchId=3298202#l99)